srml/im-online/src/lib.rs

```diff
@@ -422,3 +477,73 @@ impl<T: Trait> srml_support::unsigned::ValidateUnsigned for Module<T> {
 		TransactionValidity::Invalid(0)
 	}
 }
+
+/// An offense which is filed if a validator didn't send a heartbeat message.
+pub struct UnresponsivnessOffence<Offender> {
+	/// The current session index in which we report the unresponsive validators.
+	///
+	/// It acts as a time measure for unresponsivness reports and effectively will always point
+	/// at the end of the session.
+	session_index: SessionIndex,
+	/// The size of the validator set in current session/era.
+	validator_set_count: u32,
+	/// Authorities which were unresponsive during the current era.
```

Comment:
"Authorities that didn't send a heartbeat at the beginning of the current era."? It's confusing to me to have past tense and then "during the current era", but I understand that the heartbeat should have been sent earlier.
```suggestion
	/// Authorities that were unresponsive during the current era.
```